src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,169 @@
+                    image_streams_.push_back(img);
+                }
+                else
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_8UC3);
+                    image_streams_.push_back(img);
+                }
+            }
+            else
+            {
+                const auto& name = it.at("name").get<std::string>();
+                // streams.emplace_back(name);
+                // auto &stream = streams.back();
+                // std::cout << name << " -streams" << std::endl;
+                stream_names_.push_back(name);
+                if (name == "previewout")
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(rgb_height_, rgb_width_, CV_8UC3);
+                    image_streams_.push_back(img);
+                }
+                else if (name == "depth")
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC1);
+                    image_streams_.push_back(img);
+                }
+                else if (name == "disparity_color")
+                {
+                    std::shared_ptr<cv::Mat> img =
+                        std::make_shared<cv::Mat>(mono_height_, mono_width_, CV_16UC1);
+                    image_streams_.push_back(img);
+                }
+            }
+        }
+    }
+}
+
+DepthAI::~DepthAI() = default;
+
+// How about creating streams for everything in the vector in constructor and then updating them
+// each time get frames is called ?
+
+// TODO: Crash on size mismatch. DO this later.
+// TODO: save config as json and check if the streams are in the config and crash if they aren't
+
+void DepthAI::get_frames(std::unordered_map<std::string, CV_mat_ptr>& output_streams)
+{
+    // std::cout << "mono it-----" << mono_height_ << " ~~ " << mono_width_ << std::endl;
+
+    int count = stream_names_.size();
+    std::vector<std::string>::iterator it;
+    std::vector<bool> dirty_check(stream_names_.size(), false);
+    while (count)
+    {
+        packets_ = pipeline_->getAvailableNNetAndDataPackets(true);
+
+        for (auto sub_packet : std::get<1>(packets_))
+        {
+            it = std::find(stream_names_.begin(), stream_names_.end(), sub_packet->stream_name);
+            if (it != stream_names_.end())
+            {
+                // std::cout << "foubd it-----" << sub_packet->size() << " ~~ " <<
+                // sub_packet->stream_name << std::endl;
+                int index = it - stream_names_.begin();
+                unsigned char* img_ptr =
+                    reinterpret_cast<unsigned char*>(image_streams_[index]->data);
+                auto received_data = sub_packet->getData();
+
+                // assert()
+                memcpy(img_ptr, received_data, sub_packet->size());
```

Comment:
I feel like you need some error checking here. Is using memcpy the only way to do this? Why not make `img_ptr` a `shared_ptr` and simply dereference it?

Reply:
Not sure. have to check if it works with memcpy. Atleast form my knowledge and the way they have done in other parts thats the way.